samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -106,43 +99,30 @@ private TestRunner() {
 
   /**
    * Constructs a new {@link TestRunner} from following components
-   * @param taskClass represent a class containing Samza job logic extending either {@link StreamTask} or {@link AsyncStreamTask}
-   */
-  private TestRunner(Class taskClass) {
-    this();
-    Preconditions.checkNotNull(taskClass);
-    configs.put(TaskConfig.TASK_CLASS(), taskClass.getName());
-    this.taskClass = taskClass;
-  }
-
-  /**
-   * Constructs a new {@link TestRunner} from following components
-   * @param app samza job implementing {@link StreamApplication}
+   * @param app samza job implementing {@link SamzaApplication}
    */
-  private TestRunner(StreamApplication app) {
+  private TestRunner(SamzaApplication app) {
     this();
     Preconditions.checkNotNull(app);
     this.app = app;
   }
 
   /**
-   * Creates an instance of {@link TestRunner} for Low Level Samza Api
-   * @param taskClass samza job extending either {@link StreamTask} or {@link AsyncStreamTask}
+   * Creates an instance of {@link TestRunner} for Legacy Samza Api
+   * @param taskApp legacy samza task application
    * @return this {@link TestRunner}
    */
-  public static TestRunner of(Class taskClass) {
-    Preconditions.checkNotNull(taskClass);
-    Preconditions.checkState(
-        StreamTask.class.isAssignableFrom(taskClass) || AsyncStreamTask.class.isAssignableFrom(taskClass));
-    return new TestRunner(taskClass);
+  public static TestRunner of(LegacyTaskApplication taskApp) {
```

Comment:
Why do you need a separate API for LegacyTaskApplication? It should extend SamzaApplication too.